liblumen_alloc/src/erts/process/alloc/heap_alloc.rs

```diff
@@ -485,14 +470,19 @@ pub trait HeapAlloc {
         let tuple_ptr = unsafe { self.alloc_layout(layout)?.as_ptr() as *mut Tuple };
         let head_ptr = unsafe { tuple_ptr.add(1) as *mut Term };
         let tuple = Tuple::new(len);
+        let mut iter_len = 0;
         unsafe {
             // Write header
             ptr::write(tuple_ptr, tuple);
             // Write each element
             for (index, element) in iterator.enumerate() {
                 ptr::write(head_ptr.add(index), element);
+
+                iter_len += 1;
+                debug_assert!(index < len);
             }
         }
+        debug_assert!(iter_len == len);
```

Comment:
`wasm-pack build` is building the Rust code for release by default.  Do you want these active in the generated `.wasm`?

Reply:
I guess we should at least build with debug assertions when running tests, but I just generally insert `debug_assert`s whenever I see an invariant that must hold so that it's easier to find errors when running debug builds.

I wouldn't want debug asserts active in release builds, but I guess we should consider if we should change this into a regular `assert`?